File: facultebackend/src/main/java/tn/faculte/facultebackend/Service/ChatRoomService.java
package tn.faculte.facultebackend.Service;

import tn.faculte.facultebackend.Dto.ChatRoomDTO;
import tn.faculte.facultebackend.Entity.ChatRoom;
import tn.faculte.facultebackend.Entity.User;

import java.util.List;

public interface ChatRoomService {
    ChatRoom save(ChatRoom chatRoom);
    ChatRoom findBySenderAndRecipient(User sender, User recipient);
    List<ChatRoom> findChatRoomsByUserId(Long userId);
    ChatRoomDTO getChatRoomById(Long roomId);


    ChatRoomDTO getChatRoomByUserIds(Long loggedInUserId, Long friendId);
}

File: Faculte.tn/src/app/Services/chatroom.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { UserService } from './user.service';

@Injectable({
  providedIn: 'root'
})
export class ChatroomService {

  private baseUrl = 'https://facultetnbackend.onrender.com/api/chatrooms';

  constructor(private http: HttpClient, private userService: UserService) { }

  private getHeaders(): HttpHeaders {
    const token = this.userService.getToken();
    if (token) {
      return new HttpHeaders({
        'Authorization': `Bearer ${token}`
      });
    } else {
      console.error('No token available');
      return new HttpHeaders();
    }
  }

  getChatRoomsByUserId(userId: number): Observable<any[]> {
    const headers = this.getHeaders();
    return this.http.get<any[]>(`${this.baseUrl}/user/${userId}`, { headers });
  }

  getChatRoomById(roomId: number): Observable<any> {
    const headers = this.getHeaders();
    return this.http.get<any>(`${this.baseUrl}/${roomId}`, { headers });
  }

  getChatRoomByUserIds(loggedInUserId: number, friendId: number): Observable<any> {
    const headers = this.getHeaders();
    return this.http.get<any>(`${this.baseUrl}/user/${loggedInUserId}/friend/${friendId}`, { headers });
  }
  
}
